plugins/pdr_deterministic_plugin/ufm_sim_web_service/isolation_mgr.py

```diff
@@ -841,46 +855,53 @@ def main_flow(self):
+
+                issues = None
+                # Get telemetry data
                 if not self.test_mode:
                     self.logger.info("Retrieving telemetry data to determine ports' states")
                 else:
                     self.logger.info(f"Retrieving test mode telemetry data to determine ports' states: iteration {self.test_iteration}")
                     self.test_iteration += 1
                 try:
-                    issues = self.read_next_set_of_high_ber_or_pdr_ports()
+                    ports_counters = self.ufm_client.get_telemetry(self.test_mode)
+                    if ports_counters is None:
+                        self.logger.error("Couldn't retrieve telemetry data")
+                    else:
+                        # Detect ports to be isolated or deisolated
+                        issues = pdr_alg.read_next_set_of_high_ber_or_pdr_ports(self.ports_data, ports_counters)
                 except (KeyError,) as e:
-                    self.logger.error(f"failed to read information with error {e}")
-                if len(issues) > self.max_num_isolate:
-                    # UFM send external event
-                    event_msg = "got too many ports detected as unhealthy: %d, skipping isolation" % len(issues)
-                    self.logger.warning(event_msg)
-                    if not self.test_mode:
-                        self.ufm_client.send_event(event_msg, event_id=Constants.EXTERNAL_EVENT_ALERT, external_event_name="Skipping isolation")
-
-                # deal with reported new issues
-                else:
-                    for issue in issues.values():
-                        port = issue.port
-                        cause = issue.cause # ber|pdr|oonoc|link_down
-                        self.eval_isolation(port, cause)
-
-                # deal with ports that with either cause = oonoc or fixed
-                if self.do_deisolate:
-                    for port_state in list(self.ports_states.values()):
-                        state = port_state.get_state()
-                        cause = port_state.get_cause()
-                        # EZ: it is a state that say that some maintenance was done to the link 
-                        #     so need to re-evaluate if to return it to service
-                        if self.automatic_deisolate or cause == Constants.ISSUE_OONOC or state == Constants.STATE_TREATED:
-                            self.eval_deisolate(port_state.name)
-                ports_updated = self.update_ports_data()
-                if ports_updated:
-                    self.update_telemetry_session()
+                    self.logger.error(f"Failed to read information with error {e}")
+
+                if issues:
+                    if len(issues) > self.max_num_isolate:
+                        # UFM send external event
+                        event_msg = "got too many ports detected as unhealthy: %d, skipping isolation" % len(issues)
```

Comment:
we said this logic isn't the expected behavior. we should add the number of isolated ports in the UFM to the num of ports we want to isolate and then try to isolate as much as we can until we get to max_num_isolate, then issue the warning and skip isolation for the rest

Reply:
Yes, but is not a part of this PR. You asked not to mix different issues in one PR.
There will be additional PR for this.

Reply:
ok